src/mode/modeHandler.ts

```diff
@@ -285,9 +285,21 @@ export class ModeHandler implements vscode.Disposable {
         }
       }
     }
-    if (key === '<C-d>' && !Configuration.useCtrlKeys) {
-      key = '<D-d>';
+
+    // #2162 fixed by 5cc821e except <C-d> key due to this remapping
+    // Now keep the remapping but check if <C-d> is explicitly defined
+    // within the handleKeys scope firstly
+    if (key === '<C-d>') {
+      const useKeyCtrlD = Configuration.handleKeys['<C-d>'];
```

Comment:
This can be all simplified to one statement. 
```
if (Configuration.handleKeys['<C-d>'] ===false  || !Configuration.useCtrlKeys) {
...
}
```

Reply:
remap \<C-d\> to \<D-d\> within this statement?
what about
```
"vim.useCtrlKeys": false,
"vim.handleKeys": {
    "<C-d>": true
}
```
, should vim <C-d> enable?